### ISSUE TITLE
Hierarchical structure or inheritance of behaviors/tasks?

### ISSUE THREAD

**Issue by bruno-g** [OpenC2SIM/C2SIMArtifacts#29]
**Submitter Organization**: NMSG-145 (Operationalization of Standardized C2-Simulation Interoperability)

**Type** (Problem Report or Change Request): Problem Report
**Date of Submission**: 10 June 2022
**Product**: C2SIM XSD

**Reference**: NMSG-145 Final Report / Technical report TR-MSG-145, April 2021, Section 5.1, p92 (5-2)

Problem/Change Description:
[...] 5.1 LESSONS IDENTIFIED
The range of possible tasks (DesiredEffectCode and TaskNameCode) is significantly greater than the possible tasks or behaviors available in many of the simulations, e.g., VBS3. 
These codes have largely been derived from JC3IEDM which provides a large set of tasks so federation agreements and the mapping of C2SIM tasks to simulation capabilities need specifying carefully. 
**There could be some hierarchical structure or inheritance so that if a behavior/task is not understood by a sim, it gets rolled back to a more generic task.**

**Comment by dr-cgf-home** [comment]
Good point. If the overarching goal of C2SIM is to enable C2 systems to operate with simulations without human intervention, then the meaning of every element would (ideally} be unequivocally defined in the standard. However, many elements of the data model, and in particular tasks, represent very complex behavior with dependencies on mission, equipment, terrain, nationality/doctrine, enemy forces, etc. The military simulation community has considered the idea of defining standards for behavior for years, but without a real solution. A given simulation X can perhaps be said to "understand" the meaning of a particular task if it undergoes a rigorous validation process (for a particular use). In this case the meaning of the task term could be said to be "what simulation X does." This is all not very practical. It is probably only practical for C2SIM to provide a reasonable vocabulary at this point, and depend on "federation agreements" to provide meaning for particular uses.

As for the idea of hierarchies of behavior complexity or completeness, that is reasonable. One could express this with subclasses. For example, consider this ontology: 
![image](https://user-images.githubusercontent.com/31657895/182275440-61e252df-8aa8-46fb-9de1-bbd8b3ce5885.png)
If a task message were received that specified a Traveling task, the receiver could perhaps ignore it and select another instance of a CrossCountryTask that it understood.
